addons/hr_expense/static/src/mixins/document_upload.js

```diff
@@ -0,0 +1,35 @@
+/** @odoo-module */
+
+import { useService } from '@web/core/utils/hooks';
+
+const { useState } = owl;
+
+export const ExpenseDocumentUpload = {
+    setup() {
+        this._super();
+        this.actionService = useService('action');
+        this.orm = useService('orm');
+
+        this.state = useState({
+            selectFile: false
+        })
+    },
+
+    uploadDocument(ev) {
+        ev.preventDefault();
+        ev.stopPropagation();
+
+        this.state.selectFile = true;
+    },
+
+    async onUpload(ev) {
+        const attachmentIds = ev.map((e) => e['id']);
```

Comment:
We should probably filter first
```suggestion
        const attachmentIds = ev.filter(e => !e.error).map((e) => e['id']);
```